### PR TITLE
kubelet methods getNodeAnyWay and getHostIPAnyWay duplicate with GetNode and GetHostIP

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
remove duplicate functions

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @tallclair ptal

**Comment by hzxuzhonghu** [comment]
ping @Random-Liu @yujuhong ptal

File: pkg/kubelet/kubelet_getters.go [hunk 1]
```diff
@@ -196,18 +196,6 @@ func (kl *Kubelet) getRuntime() kubecontainer.Runtime {
 
 // GetNode returns the node info for the configured node name of this Kubelet.
 func (kl *Kubelet) GetNode() (*v1.Node, error) {
-	if kl.kubeClient == nil {
```
Comment (yujuhong) [inline pkg/kubelet/kubelet_getters.go]: `GetNode()` and `getNodeAnyWay()` are not equivalent. `getNodeAnyWay()` will return `initialNode()` if for some reason, `GetNodeInfo` returns an error. 

Could you check all the call sites to make sure they are actually interchangeable?

**Pull request closed without merging**